javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js

```diff
@@ -1,272 +1,259 @@
+  let highestCount = 0;
+  for (const num of arr) {
+    const count = counts[num];
+    if (count >= 3 && count > highestCount) {
+      highestCount = count;
+    }
+    if (count >= 4 && count > highestCount) {
+      highestCount = count;
     }
   }
 
-  isKeepScoreSuccess() {
-    const allScoreInputs = scoreInputs;
+  // Filter the original array to include only the numbers that appear the highest number of times
+  const highestDuplicates = arr.filter((num) => counts[num] === highestCount);
 
-    for (let i = 0; i < allScoreInputs.length; i++) {
-      if (allScoreInputs[i].checked) {
-        const currentValue = allScoreInputs[i].value;
-        const currentScore = this.validScoreOptions[currentValue];
-        this.totalScore += currentScore;
-        this.scoreHistory.push({ [allScoreInputs[i].value]: currentScore });
-        this.currentRound++;
-        this.rollsInCurrentRound = 0;
+  // according to rules, it should be sum of all 5 dices
+  const sumOfAllDices = displayArray.reduce((a, b) => a + b, 0);
 
-        return true;
-      }
-    }
-    alert("Please select a score.");
-    return false;
+  // Return the highest duplicates only if there are at least three or four of them
+  if (highestCount >= 4) {
+    updateRadioOption(1, sumOfAllDices);
+  } else if (highestCount >= 3) {
+    updateRadioOption(0, sumOfAllDices);
+  } else {
+    updateRadioOption(5, 0);
   }
+};
 
-  gameOver() {
-    if (
-      confirm(
-        `You scored ${game.totalScore} points. Do you want to play again?`
-      )
-    )
-      location.reload();
-  }
-}
+/**
+ * ! This will help us detect full house
+ */
+
+const detectFullHouse = (arr) => {
+  // condiiton, 3 of a kind and 2 of a kind
+  const counts = {};
 
-const game = new Game();
+  // Count the occurrences of each number in the array
+  for (const num of arr) {
+    counts[num] = counts[num] ? counts[num] + 1 : 1;
+  }
 
-rollDiceBtn.addEventListener("click", (e) => {
-  if (game.currentRound <= 6) {
-    game.rollDice();
-    game.resetRadioInputs();
-    game.generateValidScoreOptions();
-    game.enableValidScoreInputs();
-    game.updateStatsUI();
-    keepScoreBtn.disabled = false;
+  // Check if there are three of one number and two of another number
+  let hasThreeOfAKind = false;
+  let hasPair = false;
 
-    if (game.rollsInCurrentRound % 3 === 0) {
-      e.target.disabled = true;
+  for (const num in counts) {
+    if (counts[num] === 3) {
+      hasThreeOfAKind = true;
+    } else if (counts[num] === 2) {
+      hasPair = true;
     }
   }
-});
 
-keepScoreBtn.addEventListener("click", (e) => {
-  if (game.currentRound <= 6 && game.isKeepScoreSuccess()) {
-    game.resetRadioInputs();
-    game.updateStatsUI();
-    game.updateScoreUI();
-    rollDiceBtn.disabled = false;
-    e.target.disabled = true;
+  if (hasThreeOfAKind && hasPair) {
+    updateRadioOption(2, 25);
+  } else {
+    updateRadioOption(5, 0);
   }
```

Comment:
```suggestion
  if (hasThreeOfAKind && hasPair) {
    updateRadioOption(2, 25);
  };
  updateRadioOption(5, 0);
```

These shouldn't be mutually exclusive IMHO.